### PR TITLE
feat: add `isList()` and `isAssoc()` methods in ArrayHelper for array type checking

### PR DESCRIPTION
**Pull request by yassiNebeL**

### CODE REVIEW

File: src/Tempest/Support/src/ArrayHelper.php [hunk 1]
```diff
@@ -48,6 +48,30 @@ public function __construct(
...
     }
 
+    /**
+     * Determines if the array is a list.
+     *
+     * An array is a list if its keys consist of consecutive numbers.
+     *
+     * @return bool
+     */
+    public function isList()
```
Comment (innocenzi) [inline src/Tempest/Support/src/ArrayHelper.php:58]:
```suggestion
     */
    public function isList(): bool
```

File: src/Tempest/Support/src/ArrayHelper.php [hunk 2]
```diff
@@ -48,6 +48,30 @@ public function __construct(
...
+    }
+
+    /**
+     * Determines if the array is an associative.
+     *
+     * An array is associative if its keys doesn't consist of consecutive numbers.
+     *
+     * @return bool
+     */
+    public function isAssoc()
```
Comment (innocenzi) [inline src/Tempest/Support/src/ArrayHelper.php:70]:
```suggestion
     */
    public function isAssoc(): bool
```

**Review by innocenzi — CHANGES_REQUESTED** [review]

File: src/Tempest/Support/src/ArrayHelper.php (hunk 2 above)
Comment (innocenzi) [reply src/Tempest/Support/src/ArrayHelper.php]: Please remove the docblock return type as well :)

**Comment by brendt** [comment]
Thanks!

**Pull request merged**